updater/lib/dependabot/update_graph_processor.rb

```diff
@@ -96,6 +97,18 @@ def dependency_files_for(directory)
       ).returns(GithubApi::DependencySubmission)
     end
     def create_submission(source, files)
+      # return an empty submission if there are no files
+      if files.empty?
+        return GithubApi::DependencySubmission.new(
+          job_id: job.id.to_s,
+          branch: T.must(source.branch),
```

Comment:
Minor, for now this will need to be:

```suggestion
          branch: source.branch || "main",
```

I have a follow-up issue to get the branch ref from git properly, this is just a stop-gap measure for now.